lib/commands/init.js

```diff
@@ -167,6 +168,13 @@ class Init extends BaseCommand {
     try {
       const data = await input.read(() => initJson(path, initFile, this.npm.config))
       log.silly('package data', data)
+      const initPrivate = this.npm.config.get('init-private')
```

Comment:
This belongs in [`init-package-json`](https://github.com/npm/init-package-json).  We'll have to add support there first then this PR will just be adding the config flag

Reply:
https://github.com/npm/init-package-json/pull/322